packages/app-bridge/src/react/useDocumentsAndDocumentGroups.ts

```diff
@@ -0,0 +1,224 @@
+            });
+        };
+
+        const handleStandardDocumentEventUpdates = ({
+            action,
+            standardDocument,
+        }: {
+            action: EmitterAction;
+            standardDocument: Document | { id: number };
+        }) => handleEventUpdates({ action, documentOrDocumentGroup: standardDocument });
+
+        const handleLibraryDocumentEventUpdates = ({
+            action,
+            library,
+        }: {
+            action: EmitterAction;
+            library: Document | { id: number };
+        }) => handleEventUpdates({ action, documentOrDocumentGroup: library });
+
+        const handleLinkDocumentEventUpdates = ({
+            action,
+            link,
+        }: {
+            action: EmitterAction;
+            link: Document | { id: number };
+        }) => handleEventUpdates({ action, documentOrDocumentGroup: link });
+
+        const handleDocumentGroupEventUpdates = ({
+            action,
+            documentGroup,
+        }: {
+            action: EmitterAction;
+            documentGroup: DocumentGroup | { id: number };
+        }) => handleEventUpdates({ action, documentOrDocumentGroup: documentGroup });
+
+        // const handleDocumentMoveEvent = ({ action, document }: { action: 'update'; document: Document }) =>
+        //     handleEventUpdates({ action, documentOrDocumentGroup: document });
+
+        // const handleDocumentGroupMoveEvent = ({
+        //     action,
+        //     documentGroup,
+        // }: {
+        //     action: 'update';
+        //     documentGroup: DocumentGroup;
+        // }) => handleEventUpdates({ action, documentOrDocumentGroup: documentGroup });
+
+        window.emitter.on('AppBridge:GuidelineDocumentGroupAction', handleDocumentGroupEventUpdates);
+        window.emitter.on('AppBridge:GuidelineStandardDocumentAction', handleStandardDocumentEventUpdates);
+        window.emitter.on('AppBridge:GuidelineLibraryAction', handleLibraryDocumentEventUpdates);
+        window.emitter.on('AppBridge:GuidelineLinkAction', handleLinkDocumentEventUpdates);
+        // window.emitter.on('AppBridge:GuidelineDocumentMoveAction', handleDocumentMoveEvent);
+        // window.emitter.on('AppBridge:GuidelineDocumentGroupMoveAction', handleDocumentGroupMoveEvent);
+
+        return () => {
+            window.emitter.off('AppBridge:GuidelineDocumentGroupAction', handleDocumentGroupEventUpdates);
+            window.emitter.off('AppBridge:GuidelineStandardDocumentAction', handleStandardDocumentEventUpdates);
+            window.emitter.off('AppBridge:GuidelineLibraryAction', handleLibraryDocumentEventUpdates);
+            window.emitter.off('AppBridge:GuidelineLinkAction', handleLinkDocumentEventUpdates);
+            // window.emitter.on('AppBridge:GuidelineDocumentMoveAction', handleDocumentMoveEvent);
+            // window.emitter.on('AppBridge:GuidelineDocumentGroupMoveAction', handleDocumentGroupMoveEvent);
```

Comment:
```suggestion
```

Reply:
old hooks, kept just for reference and tests until the new implementation is complete